### PR TITLE
Add option to preserve digests on copy and sync

### PR DESCRIPTION
**Pull request by Jamstah**
When enabled, if digests can't be preserved an error will be raised.

Signed-off-by: James Hewitt <james.hewitt@uk.ibm.com>

### CODE REVIEW

File: docs/skopeo-copy.1.md [hunk 1]
```diff
@@ -54,6 +54,10 @@ Directory to use to share blobs across OCI repositories.
 
 After copying the image, write the digest of the resulting image to the file.
 
+**--preserve-digests**
+
+Preserve the digests during copying. Fail if the digest cannot be preserved.
```
Comment (mtrmac) [inline docs/skopeo-copy.1.md:59]: Tentative suggestion, I’m not at all sure: `--preserve-digests` is a bit of typing. Maybe `--exact`?

OTOH users who do care about this almost certainly know about digests.

@vrothberg WDYT?

**Review by mtrmac — COMMENTED** [review]
Thanks! Implementation LGTM.

File: docs/skopeo-copy.1.md (hunk 1 above)
Comment (vrothberg) [reply docs/skopeo-copy.1.md:59]: I am very unhelpful since I like both :^)

**Review by vrothberg — COMMENTED** [review]
LGTM

**Comment by Jamstah** [comment]
Rebased for conflicts.

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks for contributing!

@mtrmac @rhatdan PTanotherL

**Comment by mtrmac** [comment]
Oh — @Jamstah , could you add this to `skopeo sync` as well, please? It’s even more likely to be useful there. (Just a CLI option is fine for now.)

**Comment by mtrmac** [comment]
WRT naming, I think let’s wait for a day or so and then merge as proposed if no strong opinion comes up.

**Comment by Jamstah** [comment]
> Oh — @Jamstah , could you add this to `skopeo sync` as well, please? It’s even more likely to be useful there. (Just a CLI option is fine for now.)

Not sure what you meant by Just a CLI option, but done, just because I like you :)

I considered it for multi-arch too, but I feel like that wouldn't be useful for sync until we can specify a list of platforms we care about.

**Comment by mtrmac** [comment]
> Not sure what you meant by Just a CLI option

Exactly this (i.e. not a new field in the YAML input format). Thanks.

---

> I considered it for multi-arch too, but I feel like that wouldn't be useful for sync until we can specify a list of platforms we care about.

I agree — a single `skopeo copy` can reasonably be a building block for a script building a partial image, but `skopeo sync` works on a large number of images, so it needs to handle each single image without much external help.

**Comment by mtrmac** [comment]
Thanks again!

**Pull request merged**